misk-hibernate/src/main/kotlin/misk/hibernate/Query.kt

```diff
@@ -1,17 +1,31 @@
 package misk.hibernate
 
+import javax.persistence.criteria.CriteriaBuilder
+import javax.persistence.criteria.Root
+import javax.persistence.criteria.Selection
 import kotlin.reflect.KClass
 
 /** Base class for SQL queries. */
 interface Query<T> {
+  /** The first result to return. Defaults to 0. */
+  var firstResult: Int
+
   /** How many rows to return. Must be -1 or in range 1..10_000. */
   var maxRows: Int
 
+  /** Constrain a query using a path known only at runtime. */
```

Comment:
is it possible (slash intended) for this to be combined with static queries? Like:

```
queryFactory.newQuery(DbExample::class)
          .someQueryMethod()
          .apply { addConstraint("some_other_constraint", LT, value) }
```

or does the `rawQuery` vs `newQuery` API distinction prevent this?

Reply:
Sortof?

newQuery will fail at compile on the DbExample class. If you used rawQuery instead then the "someQueryMethod" won't exist.

You can and probably should combine static and runtime if you can though. For instance:

```
queryFactory.newQuery(QueryExample::class)
          .someQueryMethod()
          .apply { addConstraint("some_other_constraint", LT, value) }
          .listProjection(session)
```